includes/context/RequestContext.php

```diff
@@ -206,8 +206,9 @@ public function getUser() {
 		if ( $this->user === null && $wgUserForceAnon ) {
 			$this->user = new User();
 		}
+		// replace the FromToken user with a FromSession user when in a Piggyback session
+		if ( $this->user === null || PBLoginForm::isPiggyback() ) {
```

Comment:
is the PBLoginForm class always available when this function can be called?
AFAIR it's our extension that can be at least theoretically disabled, I'd suggest putting
a check if that class exists